service-config/src/main/java/com/palantir/remoting2/config/service/ProxyConfiguration.java

```diff
@@ -36,45 +36,105 @@
 @Style(visibility = Style.ImplementationVisibility.PACKAGE, builder = "new")
 public abstract class ProxyConfiguration {
 
+    enum Type {
+
+        /**
+         * Use a direct connection. This option will bypass any JVM-level configured proxy settings.
+         */
+        direct,
+
+        /**
+         * Use an http-proxy specified by {@link ProxyConfiguration#hostAndPort()}  and (optionally)
+         * {@link ProxyConfiguration#credentials()}.
+         */
+        http;
+    }
+
     /**
      * The hostname and port of the HTTP/HTTPS Proxy. Recognized formats include those recognized by {@link
      * com.google.common.net.HostAndPort}, for instance {@code foo.com:80}, {@code 192.168.3.100:8080}, etc.
      */
-    public abstract String hostAndPort();
+    @JsonProperty("hostAndPort")
```

Comment:
do you mean *de*serializing the camel-cased hostAndPort? I don't think that was ever supported because of the annotation `@JsonDeserialize(builder = ProxyConfiguration.Builder.class)`